docs/source/use_cases/run_picard_and_gatk/index.rst

```diff
@@ -17,20 +17,178 @@ __ RenderedVersion_
+
+We use a test readset here from `genomics-test-data <https://pantheon.corp.google.com/project/genomics-test-data/storage/browser/gatk-tools-java/>`_ project. 
+
+Specifying a genomics region to use from the readset
+----------------------------------------------------
+The INPUT urls are of the form **https://<GA4GH provider>/readgroupsets/<readgroupset id>[/sequence][/start-end]**.
+
+For example:
+
+.. code-block:: shell
+
+  java -jar dist/picard.jar ViewSam \ 
+  INPUT=https://www.googleapis.com/genomics/v1beta2/readgroupsets\
+  /CMvnhpKTFhD3he72j4KZuyc/chr17/41196311-42677499 \
+  GA4GH_CLIENT_SECRETS=../client_secrets.json
+
+
+Timing the reading speed from the cloud
+---------------------------------------
+You can run `gatk-tools-java/src/main/scripts/example.sh <https://github.com/googlegenomics/gatk-tools-java/blob/master/src/main/scripts/example.sh>`_ with and without "grpc" command line parameter to see the difference in reading speed. The timing statistics are dumped to the terminal.
+We benchmarked **x11** speed improvements with GRPC compared to REST, giving **~12,000 reads/second**.
+
+The tests were done on `Platinum Genomes NA12877_S1.bam dataset <https://pantheon.corp.google.com/storage/browser/genomics-public-data/platinum-genomes/bam/?_ga=1.197206447.160385476.1431305548>`_, please see the `detailed writeup of the test procedure and results <https://docs.google.com/document/d/1Br7RMSbAChNpG6pi2teujf-YthczF-rAM1afSgDoCgQ/edit#>`_ if you want to repeat the test.
+
+We therefore recommend running GRPC variants of command line.
+
+Other Picard tools you can run
+------------------------------
+You can run MarkDuplicates or MarkDuplicatesWithMateCigar tools like this:
+
+.. code-block:: shell
+
+  java \
+    -Xbootclasspath/p:../gatk-tools-java/lib/alpn-boot-8.1.3.v20150130.jar \
+    -Dga4gh.using_grpc=true \
+    -jar dist/picard.jar MarkDuplicates \
+    INPUT=https://www.googleapis.com/genomics/v1beta2/readgroupsets/CK256frpGBD44IWHwLP22R4/ \
+    OUTPUT=output.bam \
+    METRICS_FILE=output.metrics \
+    GA4GH_CLIENT_SECRETS=../client_secrets.json
+
+Figuring out a url for your dataset
+-----------------------------------
+In the examples above we have been using urls of the form https://www.googleapis.com/genomics/v1beta2/readgroupsets/XXX where XXX is the id of the readset. 
+
+How do you find an ID of the readset from the `Discover Published Data`_ set or from your own project ? 
+
+We will do it step by step using the command line API client. 
+
+* Lets say we want to use `Platinum Genomes NA12877_S1.bam readgroupset <https://pantheon.corp.google.com/storage/browser/genomics-public-data/platinum-genomes/bam/?_ga=1.197206447.160385476.1431305548>`_ from `genomics-public-data <https://cloud.google.com/genomics/data/1000-genomes?hl=en>`_ project.
+
+* The `documentation <https://cloud.google.com/genomics/data/1000-genomes?hl=en>`_ page states that the dataset id for this set of files is **10473108253681171589**.
```

Comment:
For consistency, point to :doc:`/use_cases/discover_public_data/1000_genomes` instead of https://cloud.google.com/genomics/data/1000-genomes?hl=en